File: src/main/java/org/kane/domain/service/recipe_recource/ingredient/IngredientService.java
package org.kane.domain.service.recipe_recource.ingredient;

import org.kane.database.entity.Recipe;
import org.kane.database.entity.recipe_recource.Ingredient;
import org.kane.domain.DTO.entityDTO.diary.recipe_recource.ingredient.IngredientChangeDTO;
import org.kane.domain.DTO.entityDTO.diary.recipe_recource.ingredient.IngredientCreateDTO;
import org.kane.domain.DTO.entityDTO.diary.recipe_recource.ingredient.IngredientEditDTO;
import org.kane.domain.DTO.entityDTO.diary.recipe_recource.ingredient.IngredientShowDTO;

import java.util.List;

public interface IngredientService {
    Ingredient createIngredient(IngredientCreateDTO ingredientCreateDTO, Recipe recipe);
    Ingredient updateIngredient(IngredientEditDTO ingredient);
    void removeIngredient(Long id);
    List<IngredientShowDTO> getShowIngredients(Long recipeID);
    IngredientShowDTO toggleMeasureUnit(IngredientChangeDTO ingredientChangeDTO);
}

File: client/src/app/service/recipe_resource/ingredient.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Endpoint} from '../../util/endpoint';
import {Observable} from 'rxjs';
import IngredientShowDTO from '../../DTO/entity_dto/recipe-recource/ingredient/ingredient-show.dto';
import IngredientChangeDTO from '../../DTO/entity_dto/recipe-recource/ingredient/ingredient-change.dto';

@Injectable({
  providedIn: 'root',
})
export default class IngredientService {
  private readonly http = inject(HttpClient);
  private readonly ingredientAPI = new Endpoint('ingredient');

  removeIngredient(id: number):Observable<any>{
    return this.http.delete(
      this.ingredientAPI.builder()
        .points("delete", id.toString())
        .build());
  }

  getShowIngredients(id:number):Observable<IngredientShowDTO[]>{
    return this.http.get<IngredientShowDTO[]>(
      this.ingredientAPI.builder()
        .points("get", id.toString())
        .build());
  }

  toggleMeasureUnit(ingredient:IngredientChangeDTO):Observable<IngredientShowDTO>{
    return this.http.post<IngredientShowDTO>(
      this.ingredientAPI.builder()
        .points("toggle")
        .build(),
      ingredient)
  }
}
